### PR TITLE
Add Networking Port Support for AI vs AI

### LINKED ISSUE #1666: Add networking setup to documentation to run Simulator
### Description of the task

While trying to run the Simulator with 2 AIs, we realized by Ubuntu closes all ports by default (no data goes through), so we need to explicitly open the ports we use to communicate in order for things to work.

Eg.
```
sudo ufw allow 42069
sudo ufw allow 42070
sudo ufw allow 42071
```

We should figure out how we want to involve this in our environment setup. Options:
* Add it to the documentation
* Add it to the setup scripts
* Something else?

In all cases it's a bit tricky since we'd have to make sure the values are up to date with our internal constants, or tell the user to check the contants file for the most up to date values. Thoughts on the best way to do this?

### Acceptance criteria

- [ ] Figure out how to set up networking ports in our environment setup
- [ ] Do it

### PR DESCRIPTION
**Pull request by kelvinkoon**
### Description

Quick changes to open ports 42069 to 42073 for `ai vs ai`. 
In testing, most people were able to get `ai vs ai` running but in the odd case, the ports need to be explicitly opened.
Documentation in the `Getting Started` section was also updated.

### Testing Done

Running the Bash script gave a status that was expected.

### Resolved Issues

Resolves #1666 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by MathewMacDougall — COMMENTED** [review]
This could also be run as part of the setup scripts. Otherwise LGTM.

I'm assuming `ufw` works on WSL as well? Not sure if that's being fully supported or not, just pointing it out in case.

**Comment by kelvinkoon** [comment]
> This could also be run as part of the setup scripts. Otherwise LGTM.
> 
> I'm assuming `ufw` works on WSL as well? Not sure if that's being fully supported or not, just pointing it out in case.

Going to require someone on WSL to test. Regardless, I'll make the documentation more explicitly mention you need to open the ports instead of simply leaving it at `run ai vs ai setup`

**Comment by kelvinkoon** [comment]
> This could also be run as part of the setup scripts. Otherwise LGTM.
> 
> I'm assuming `ufw` works on WSL as well? Not sure if that's being fully supported or not, just pointing it out in case.

Moved the script to the setup scripts directory.

File: docs/getting-started.md [hunk 1]
```diff
@@ -190,6 +190,7 @@ Now that you're setup, if you can run it on the command line, you can run it in
 ### Running AI vs AI
 1. Open your terminal, `cd` into `Software/src`
 2. Run `./software/run_ai_vs_ai.sh interface_name`, using the same interface as from [above](#running-our-ai-simulator-or-robot-diagnostics)
+3. If AI vs AI is not running, the networking ports are likely the cause. Run `./../../environment_setup/configure_ai_vs_ai.sh` to open the ports.
```
Comment (MathewMacDougall) [inline docs/getting-started.md]: nit: I think giving the aboluste path for `configure_ai_vs_ai.sh` will be more clear.

File: environment_setup/configure_ai_vs_ai.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+# The name of this script
+THIS_SCRIPT_FILENAME=$(basename "$0")
```
Comment (MathewMacDougall) [inline environment_setup/configure_ai_vs_ai.sh]: This variable is unused?

**Review by MathewMacDougall — COMMENTED** [review]
quick comments

File: environment_setup/configure_ai_vs_ai.sh (hunk 2 above)
Comment (kelvinkoon) [reply environment_setup/configure_ai_vs_ai.sh]: Removed.

File: docs/getting-started.md (hunk 1 above)
Comment (kelvinkoon) [reply docs/getting-started.md]: Addressed.

File: docs/getting-started.md [hunk 3]
```diff
@@ -190,6 +190,7 @@ Now that you're setup, if you can run it on the command line, you can run it in
 ### Running AI vs AI
 1. Open your terminal, `cd` into `Software/src`
 2. Run `./software/run_ai_vs_ai.sh interface_name`, using the same interface as from [above](#running-our-ai-simulator-or-robot-diagnostics)
+3. If AI vs AI is not running, the networking ports are likely the cause. `cd` into `Software/environment_setup` and run `./configure_ai_vs_ai.sh` to open the ports.
```
Comment (jonathanlew) [inline docs/getting-started.md]: What is the failure mode when the ports aren't allowed? Can we be more specific?

**Review by jonathanlew — CHANGES_REQUESTED** [review]
Has this been validated on a machine that has this problem?

File: docs/getting-started.md (hunk 3 above)
Comment (kelvinkoon) [reply docs/getting-started.md]: Specified that the visualizer won't run (ie. nothing happens when you run a play)

**Comment by kelvinkoon** [comment]
> Has this been validated on a machine that has this problem?

The issue is Mat's computer has been upgraded to Ubuntu 20 and there's seemingly no exact way to replicate the issue b/c no one on the team has issues. I noticed on my laptop enabling `UFW` would cause the visualizer to freeze and my hunch is `UFW` is all incoming traffic unless specified. Since this problem is *relatively* uncommon, there's been some consensus that disabling UFW is a good enough solution (for now). Any issues with this conclusion? @akhilveeraghanta @jonathanlew @MathewMacDougall

File: environment_setup/configure_ai_vs_ai.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
Comment (akhilveeraghanta) [inline environment_setup/configure_ai_vs_ai.sh]: maybe just worth calling this script `disable_ufw.sh` so its clear what the user is getting themselves into

**Review by akhilveeraghanta — CHANGES_REQUESTED** [review]
lgtm! just one comment

File: environment_setup/configure_ai_vs_ai.sh (hunk 4 above)
Comment (kelvinkoon) [reply environment_setup/configure_ai_vs_ai.sh]: Agreed. Addressed.

File: docs/getting-started.md [hunk 5]
```diff
@@ -190,6 +190,7 @@ Now that you're setup, if you can run it on the command line, you can run it in
 ### Running AI vs AI
 1. Open your terminal, `cd` into `Software/src`
 2. Run `./software/run_ai_vs_ai.sh interface_name`, using the same interface as from [above](#running-our-ai-simulator-or-robot-diagnostics)
+3. If AI vs AI is not running properly (eg. the visualizer does not respond after starting a play), closed ports are likely the cause. `cd` into `Software/environment_setup` and run `./disable_ufw.sh` to disable UFW and allow traffic through the visualizer ports.
```
Comment (jonathanlew) [inline docs/getting-started.md]: it really doesn't seem worth it to make whole script for one command

**Review by jonathanlew — CHANGES_REQUESTED** [review]
I agree with the conclusions and adding a little comment to getting-started guide

File: docs/getting-started.md (hunk 5 above)
Comment (kelvinkoon) [reply docs/getting-started.md]: Changed to just say "Run `sudo ufw disable`"

**Review by jonathanlew — APPROVED** [review]

**Comment by MathewMacDougall** [comment]
Agree. I suspect that `ufw` is disabled by default (it is on my fresh ubuntu 20 install), and I previously had it enabled for something else, which caused the initial problems. So I think leaving it alone (or explicitly disabling it) is a better solution. I was an anomaly.

**Review by akhilveeraghanta — APPROVED** [review]

**Pull request merged**